src/layout/Payment/PaymentComponent.tsx

```diff
@@ -45,54 +45,56 @@ export const PaymentComponent = ({ node }: PropsFromGenericComponent<'Payment'>)
     }
   }, [paymentInfo, next]);
 
-  if (busy || paymentDoesNotExist) {
+  if ((busy || paymentDoesNotExist) && !paymentError) {
     return <SkeletonLoader />;
   }
 
   return (
     <ComponentStructureWrapper node={node}>
-      <PaymentDetailsTable
-        orderDetails={paymentInfo?.orderDetails}
-        tableTitle={title}
-        description={description}
-      />
-      <div>
-        {paymentInfo?.status === PaymentStatus.Failed && (
-          <Alert severity='warning'>
-            <Lang id='payment.alert.failed' />
-          </Alert>
-        )}
-        {paymentInfo?.status === PaymentStatus.Paid && (
-          <Alert severity={'info'}>
-            <Lang id='payment.alert.paid' />
-          </Alert>
-        )}
-      </div>
-      <div className={classes.buttonContainer}>
-        {paymentInfo?.status === PaymentStatus.Created && (
-          <>
+      <div className={classes.paymentContainer}>
+        <PaymentDetailsTable
+          orderDetails={paymentInfo?.orderDetails}
+          tableTitle={title}
+          description={description}
+        />
+        <div className={classes.alertContainer}>
+          {(paymentInfo?.status === PaymentStatus.Failed || paymentError) && (
+            <Alert severity='warning'>
+              <Lang id='payment.alert.failed' />
+            </Alert>
+          )}
+          {paymentInfo?.status === PaymentStatus.Paid && (
+            <Alert severity={'info'}>
+              <Lang id='payment.alert.paid' />
+            </Alert>
+          )}
+        </div>
+        <div className={classes.buttonContainer}>
+          {(paymentInfo?.status === PaymentStatus.Created || paymentError) && (
```

Comment:
Does the payment status change if the payActionMutation fails?

Reply:
No, the PaymentStatus is referring to the payment "instance"s status, not the payment action itself if i have understood it correctly. The status will only be changed by the payment action if it succeeds and sets it to "Paid". Otherwise it will first be "Uninitialized" untill the payment instance has been attempted to be created which will result in "Created" if it succeeds. The status "Failed" does not seem to be used in back-end @bjorntore?

Reply:
Cause if the status does not change if it fails, do you need the `|| paymentError`?

Reply:
It would work to instead check if the status is uninitialized, however i think we are more sure to provide the buttons if something fails if we check the error. The payment can fail for a multitude of reasons, not only if the paymentinstance has not been created. So regardless of the payment instances status, we would like to provide options to try again or go back if the payment action fails.